### ISSUE TITLE
What is Rollbax supposed to report when enable_crash_reports = true?

### ISSUE THREAD

**Issue by messutied** [elixir-addicts/rollbax#96]
I'm sorry but I'm not been able to get Rollbax to report anything automatically when setting enable_crash_reports = true, perhaps I don't fully understand constitutes a crash in elixir, I'm trying to test it simply by throwing an exception, or doing a miss match like `{1} = {2}` or the `DoesNotExist.for_sure()` without try/catch, my app is crashing but nothing gets reported.

With the manual try/catch `Rollbax.report(:error, exception, System.stacktrace())` it is working, but I can't see what `enable_crash_reports: true` does.

Perhaps someone can guide me on how I should be using Rollbax out of a phoenix app (in phoenix it's just like magic), should I be using try/catch and report manually? or is `enable_crash_reports: true` supposed to catch all/most the errors for me?

Thanks a lot.

**Comment by whatyouhide** [comment]
@messutied `:enable_crash_reports` is for reporting crashes that happen in processes like GenServers, Tasks, and so on. It doesn't catch "exceptions" automatically, only if those exceptions crash a process like the ones I just mentioned. Where are you raising the exceptions?

**Comment by messutied** [comment]
Thanks for the quick reply @whatyouhide, ah I see, I was raising the exception in a supervisor, it crashed the whole Application and somehow I thought this was in the "GenServers, Tasks, and so on" category... I think I get it now, I'll try raising the exception in a GenServer, thanks a lot.

**Comment by whatyouhide** [comment]
Awesome! Closing this now then. Feel free to reopen if that doesn't work.

**Comment by adamu** [comment]
I'm also confused by this. Should running the following code in iex result in a Rollbax report?

```elixir
spawn(fn -> raise "aaaah" end)
```

From what I can see, it doesn't, although the process clearly crashed.

**Comment by Cellane** [comment]
I… think that the standard reporter (no longer?) properly supports regular crashes. If I’m not mistaken, the flow goes something like this, right?

1. When `Rollbax` is started, it registers `Rollbax.Logger` as a error report handler [here](https://github.com/ForzaElixir/rollbax/blob/v0.11.0/lib/rollbax.ex#L89) (if `enable_crash_reports: true`)
2. `Rollbax.Logger` is a `gen_event` behaviour and receives errors via `handle_event/2`; for each event, it runs them through a list of reporters, which by default include `Rollbax.Reporter.Standard`, but [can be configured separately](https://github.com/ForzaElixir/rollbax/blob/v0.11.0/lib/rollbax/logger.ex#L56)
3. `Rollbax.Reporter.Standard` attempts to pattern match on various error types and transform them into `Rollbax.Exception` structs
4. In the case of a simple process crash that @adamu mentioned above, the `handle_error_format/2` is called with two parameters:
    - A charlist with error message: `'Error in process ~p on node ~p with exit value:~n~p~n'`
    - A list of three elements, `pid(), atom(), tuple()`: `[#PID<0.2957.0>, :censored@89401c43b4b7, {%RuntimeError{message: "aaaah"}, [{:elixir_eval, :__FILE__, 1, [file: 'iex', line: 75]}]}]}`
5. There is a clause that almost looks like it should match [here](https://github.com/ForzaElixir/rollbax/blob/v0.11.0/lib/rollbax/reporter/standard.ex#L82-L93), but it matches on lists of two elements, not three, thus it falls through [to the clause on line 97](https://github.com/ForzaElixir/rollbax/blob/v0.11.0/lib/rollbax/reporter/standard.ex#L97) which just ignores the event

In other words, a minimal fix should be:

```diff
-  defp handle_error_format('Error in process ' ++ _, [pid, {reason, stacktrace}]) do
+  defp handle_error_format('Error in process ' ++ _, [pid, _node, {reason, stacktrace}]) do
```

With that change in the reporter, once I re-run Adam’s code, I can see `04:08:08.095 [info] (Rollbax) registered report.` in the IEx shell.
